subql_wika/src/mappings/eventHandlers.ts

```diff
@@ -0,0 +1,163 @@
+import {SubstrateExtrinsic, SubstrateEvent, SubstrateBlock} from "@subql/types";
+import {PluginPostgres} from "../plugins/postgres";
+import {PluginNeo4j} from "../plugins/neo4j";
+import {PluginElasticSearch} from "../plugins/elastic_search";
+import {fetchMetadata} from "../plugins/page_metadata";
+import crypto from 'crypto';
+import {PluginEmails} from '../plugins/emails' ;
+
+
+
+/**
+* Holds instances of the data access plugins
+* and the ETL logic for the events emitted by the Wika blockchain:
+* LikeEvent
+* UrlRegisteredEvent
+* @remarks
+* Logging is directed to the global logger if available, otherwise to console.
+*/
+export class EventHandlers {
+
+    // Data Plugins
+    private postgres: PluginPostgres ;
+    private neo4j: PluginNeo4j ;
+    private elastic: PluginElasticSearch ;
+    private emails: PluginEmails ;
+
+    // Logger
+    private log ;
+
+    /**
+    * Constructor relies on env vars
+    * for the list of env vars, see the docs of each Plugin, or the docker-compose file
+    */
+    constructor() {
+        // Instantiate data plugins
+        this.postgres = new PluginPostgres() ;
```

Comment:
1. This is not mandatory, but there is a shorthand for typescript members and their intitialization:
    ```typescript
    // option 1: defining the member within the constructor signature and assign a default value 
    constructor (public postgress = new PluginPostgres())
    
    // option 2: define the member completely outside the constructor and assign a default value
    public postgress = new PluginPostgres()
    constructor() {}
    ```
    Both these options should work, option 1 has the advantage that you can overwrite the members when constructing the object. Feel free to use one of these shorthands to make your code less redundant.

2. Just wondering, do you leave a blank space b/w semicolon `;` and the last symbol `PluginPostgres()` on purpose? I've never seen this style in any JS/TS project.

Reply:
1. Went with option 1, thanks for the trick, could be helpful for testing...
2. I do Python, Rust, Java and Javascript lately so it's hard to keep up with all the different conventions. But added prettier to the rescue so `yarn format` should take care of all those details now :)